integration-test/src/test/java/org/airsonic/test/cucumber_hooks/docker/DynamicDockerHook.java

```diff
@@ -118,10 +120,16 @@ public void startServer() {
                 String ipAddress = next.getValue().ipAddress();
                 serverUri = "http://" + ipAddress + ":" + dockerPort;
             } catch(Exception e) {
+                isException = true;
                 throw new RuntimeException("Could not determine container ("+name+") address", e);
             }
         } catch (DockerException | InterruptedException e) {
+            isException = true;
             throw new RuntimeException(e);
+        } finally {
+            if(isException){
+                stopServer();
```

Comment:
Can't this be run in all cases anyway?

Reply:
Please correct me if I am wrong, but if stopServer is run when an exception doesn't occur, it would clean up the instance before the tests can be ran, isn't it?

Reply:
Righty-right